include/darwin/cdrom.h

```diff
@@ -0,0 +1,717 @@
+/*
+ * -- <darwin/cdrom.h>
+ * General header file for darwin CD-ROM drivers 
+ * Copyright (C) 1992         David Giller, rafetmad@oxy.edu
+ *               1994, 1995   Eberhard Moenkeberg, emoenke@gwdg.de
+ *               1996         David van Leeuwen, david@tm.tno.nl
+ *               1997, 1998   Erik Andersen, andersee@debian.org
+ *               1998-2000    Jens Axboe, axboe@suse.de
+ */
+ 
+#ifndef	_DARWIN_CDROM_H
+#define	_DARWIN_CDROM_H
+
+#include "darwin/types.h"
+
+/*******************************************************
+ * All Darwin CD-ROM application programs will use this 
+ * (and only this) include file.  It is my hope to provide Darwin with
+ * a uniform interface between software accessing CD-ROMs and the various 
+ * device drivers that actually talk to the drives.  There may still be
+ * 23 different kinds of strange CD-ROM drives, but at least there will 
+ * now be one, and only one, Darwin CD-ROM interface.
+ *
+ * Additionally, all Darwin application programs 
+ * should use the O_NONBLOCK option when opening a CD-ROM device 
+ * for subsequent ioctl commands.  This allows for neat system errors 
+ * like "No medium found" or "Wrong medium type" upon attempting to 
+ * mount or play an empty slot, mount an audio disc, or play a data disc.
+ * Generally, changing an application program to support O_NONBLOCK
+ * is as easy as the following:
+ *       -    drive = open("/dev/cdrom", O_RDONLY);
+ *       +    drive = open("/dev/cdrom", O_RDONLY | O_NONBLOCK);
+ * It is worth the small change.
+ *
+ *******************************************************/
+
+/* When a driver supports a certain function, but the cdrom drive we are 
+ * using doesn't, we will return the error EDRIVE_CANT_DO_THIS.  We will 
+ * borrow the "Operation not supported" error from the network folks to 
+ * accomplish this.  Maybe someday we will get a more targeted error code, 
+ * but this will do for now... */
+#define EDRIVE_CANT_DO_THIS  EOPNOTSUPP
```

Comment:
Where is this `EDRIVE_CANT_DO_THIS` used? I do not see place where is needed this nor any other macros. So please do not include and bundle dead code.

Reply:
As per other comments of mine, the alternative is to prune (Linux-verbatim) code to the minimum, of course it could make it life more difficult to reintroduce defines/etc later down the road, if needed by new (core) code in udftools.

Anyway, albeit we're talking about three/four self-contained files almost verbatim from /usr/include in Linux, if you feel like this is absolutely needed, please let me know, and I'll take care of the pruning.